src/main/java/com/snowflake/kafka/connector/internal/FileNameUtils.java

```diff
@@ -192,4 +195,46 @@ private static String readFromFileName(String fileName, int index) {
 
     return matcher.group(index);
   }
+
+  /**
+   * construct a log string that contains offset-range of input files
+   *
+   * @param files list of files
+   * @param fileType type of files processed (ex loadedFiles)
+   * @param shouldLogDebugOrTrace should Debug or Trace log be added
+   * @return string that must be logged
+   */
+  static String prepareFilesOffsetsLogString(
+      List<String> files, String fileType, boolean shouldLogDebugOrTrace
+  ) {
+    StringBuilder logString = new StringBuilder();
+    logString.append(", ").append(fileType).append(" offset range: [");
+    long[][] offsetArray = new long[files.size()][2];
+    String file;
+    for (int i =0; i < files.size(); i++) {
+      file = files.get(i);
+      offsetArray[i][0] = FileNameUtils.fileNameToStartOffset(file);
+      offsetArray[i][1] = FileNameUtils.fileNameToEndOffset(file);
+    }
```

Comment:
we could use java streams for such operations and some pair/entry to avoid 2d arrays